src/main/java/org/isf/menu/gui/Menu.java

```diff
@@ -104,6 +110,52 @@ public static boolean isAtLeastVersion(String requiredVersion) {
 		}
 	}
 
+	public static void setUIFont(javax.swing.plaf.FontUIResource f) {
+		Enumeration<Object> keys = UIManager.getDefaults().keys();
+		while (keys.hasMoreElements()) {
+			Object key = keys.nextElement();
+			Object value = UIManager.get(key);
+			if (value instanceof javax.swing.plaf.FontUIResource)
+				UIManager.put(key, f);
+		}
+	}
+
+	public static void fontChecker() {
+
+		GeneralData.initialize();
+		MessageBundle.getBundle();
+		String textToCheck = MessageBundle.getMessage("angal.common.address.txt"); // Any string from bundle to check
+		LOGGER.debug("textToCheck: {}", textToCheck);
+
+		Font currentFont = UIManager.getFont("Label.font");
+
+		if (currentFont != null) {
+			LOGGER.debug("Current Font: " + currentFont.getFontName() + " " + currentFont.getSize());
+		} else {
+			LOGGER.debug("Unable to retrieve the current font from the L&F.");
+			return;
+		}
+
+		// Check if the currentFont can display the textToCheck
+		if (currentFont.canDisplayUpTo(textToCheck) == -1) {
+			LOGGER.debug("The current font supports the selected language.");
+		} else {
+			LOGGER.debug("The current font does not support the selected language.");
+
+			// Find a font that supports the provided textToCheck
+			Font[] availableFonts = GraphicsEnvironment.getLocalGraphicsEnvironment().getAllFonts();
+			for (Font font : availableFonts) {
+				if (font.canDisplayUpTo(textToCheck) == -1) {
+					LOGGER.debug("Found a font that supports the selected language: {}", font.getFontName());
+					setUIFont(new javax.swing.plaf.FontUIResource(font.getFontName(), Font.PLAIN, 12));
```

Comment:
the class should be imported and not fully qualified in the code

Reply:
right, thanks!